### PR TITLE
feat(artist-creation): auto-persist artists, unnest bulk inserts, Follow RPC wiring

### PR DESCRIPTION
**Pull request by pannpers**
## 🔗 Related Issue

Refs: #artist-creation

## 📝 Summary of Changes

`ListTop` and `ListSimilar` RPCs returned `Artist` entities with empty `id` fields because artists fetched from Last.fm were never persisted locally. This caused the entire follow → concert pipeline to be non-functional.

### Changes

**Entity layer**
- Add `entity.NewArtist(name, mbid string) *Artist` constructor with auto-generated UUIDv7 ID
- Remove unused `CreateTime` field from `Artist` struct
- Extend `ArtistRepository.Create` to variadic bulk-upsert: `Create(ctx, artists ...*Artist) ([]*Artist, error)`

**Repository layer**
- `ArtistRepository.Create`: bulk insert via PostgreSQL `unnest` pattern with `ON CONFLICT (mbid) DO NOTHING`; returns all artists (new + pre-existing) with valid IDs
- `ConcertRepository.Create`: refactor from manual `fmt.Sprintf` placeholder construction to `unnest` pattern (removes 65,535 param limit and batching loop)
- DB migrations: add UNIQUE INDEX on `artists.mbid`; drop unused `created_at`/`updated_at` from artists table

**Use case layer**
- `ListTop` and `ListSimilar` now call `ArtistRepository.Create` after fetching from Last.fm, ensuring every returned `Artist` has a valid `id`

**Adapter / infra layer**
- Fix `Create` RPC handler bug: `req.Msg.Mbid` was silently dropped — now correctly passed to use case
- Update LastFM and MusicBrainz adapters to use `entity.NewArtist()` constructor

**Tests**
- Table-driven integration test for `ArtistRepository.Create` (all edge cases incl. deduplication)
- Unit tests for `ArtistHandler` and `ArtistUseCase` (ListTop/ListSimilar auto-persist paths)

## 📋 Commit Log

```
9ff7a95 test: add and update tests for artist-creation changes
1a86554 fix(adapter): pass MBID to use case in Create handler; use NewArtist
b2cb131 feat(usecase): auto-persist external artists in ListTop and ListSimilar
db49aca refactor(rdb): rewrite ConcertRepository.Create using unnest pattern
66ed472 feat(rdb): implement ArtistRepository.Create with unnest bulk upsert
d9d94fa chore(mocks): regenerate ArtistRepository mock for new Create signature
75d5059 refactor(entity): add NewArtist constructor and remove CreateTime field
```

## ✅ Self-Checklist

- [x] I have linked the related issue.
- [x] I have added or updated tests to cover my changes.
- [ ] I have updated the relevant documentation (e.g., `README.md`, `CLAUDE.md`).
- [x] I have run linters locally and all checks have passed.
- [x] My code follows the architecture and style guidelines of the project.

### CODE REVIEW

File: internal/infrastructure/database/rdb/concert_repo.go [hunk 1]
```diff
@@ -15,12 +14,16 @@ type ConcertRepository struct {
...
 
 const (
-	// maxConcertsPerBatch limits the number of concerts inserted in a single SQL statement.
-	// PostgreSQL has a maximum of 65,535 parameters per statement.
-	// Each concert uses 8 parameters (events table) + 2 parameters (concerts table) = 10 total.
-	// To stay well within limits and maintain readability, we batch at 500 concerts.
-	maxConcertsPerBatch = 500
-
+	insertEventsUnnestQuery = `
+		INSERT INTO events (id, venue_id, title, listed_venue_name, local_event_date, start_at, open_at, source_url)
```
[the remark this answers is not in the document]
Comment (pannpers) [reply internal/infrastructure/database/rdb/concert_repo.go]: ✅ This was already addressed in f152e19.

This was caught independently when CI tests failed with `column "start_at" is of type timestamp with time zone but expression is of type time without time zone (SQLSTATE 42804)`. Fixed by changing the unnest cast from `::time[]` to `::timestamptz[]` to match the column type.

File: internal/entity/artist.go [hunk 2]
```diff
@@ -21,12 +20,18 @@ type Artist struct {
...
+func NewArtist(name, mbid string) *Artist {
+	return &Artist{
+		ID:   newID(),
+		Name: name,
+		MBID: mbid,
+	}
+}
+
+func newID() string {
 	id, _ := uuid.NewV7()
```
[the remark this answers is not in the document]
Comment (pannpers) [reply internal/entity/artist.go]: Thanks for the suggestion!

After reviewing, I've decided not to implement this change because `uuid.NewV7()` can only fail if the OS entropy source itself is broken (i.e., `crypto/rand.Read` fails), which is an unrecoverable system-level condition regardless. More importantly, this codebase uses UUIDv7's time-ordered property for implicit temporal ordering after `created_at` was removed from the artists table. Falling back to UUIDv4 would silently break that ordering guarantee. The current behavior — returning a zero-value UUID on the vanishingly unlikely failure — results in an immediate PK violation at the DB layer, which is an explicit, detectable error. Open to further discussion if you see it differently.

File: internal/infrastructure/database/rdb/concert_repo.go [hunk 3]
```diff
@@ -66,71 +69,50 @@ func (r *ConcertRepository) ListByArtist(ctx context.Context, artistID string, u
...
+	listedVenueNames := make([]*string, n)
+	eventDates := make([]time.Time, n)
+	startTimes := make([]*time.Time, n)
+	openTimes := make([]*time.Time, n)
+	sourceURLs := make([]string, n)
+	artistIDs := make([]string, n)
+
+	for i, c := range concerts {
+		if c == nil {
+			continue
```
[the remark this answers is not in the document]
Comment (pannpers) [reply internal/infrastructure/database/rdb/concert_repo.go]: ✅ Addressed in 6363c0b.

**What was changed**: Replaced the index-based \`continue\` pattern with an upfront compaction pass. All nil elements are filtered into a separate \`valid\` slice first; the target arrays (\`eventIDs\`, \`venueIDs\`, etc.) are then allocated with \`len(valid)\` and populated from \`valid\` without gaps. A regression test ("nil elements are skipped without DB error") was added in 0178caa.

**Why**: The previous nil guard used \`continue\` inside the index loop, leaving zero-value holes at position \`i\` in the pre-allocated arrays. PostgreSQL receives an empty string for that UUID slot and rejects it with \`invalid input syntax for type uuid: ""\`. Compacting first ensures the arrays are always dense and correctly sized.
